### PR TITLE
[SPARK-28483][FOLLOW-UP] Fix flaky test in BarrierTaskContextSuite

### PR DESCRIPTION
**Pull request by WeichenXu123**
### What changes were proposed in this pull request?

I fix the test "barrier task killed" which is flaky:

* Split interrupt/no interrupt test into separate sparkContext. Prevent them to influence each other.
* only check exception on partiton-0. partition-1 is hang on sleep which may throw other exception.

### Why are the changes needed?
Make test robust.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by WeichenXu123** [comment]
Jenkins, retest this please.

File: core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala [hunk 1]
```diff
@@ -201,15 +201,25 @@ class BarrierTaskContextSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
-  test("barrier task killed") {
+  test("barrier task killed, no interrupt") {
+    val conf = new SparkConf()
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala]: Can this part also go into the helper method to not duplicate it or is it used elsewhere?

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by WeichenXu123** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by HyukjinKwon** [comment]
@WeichenXu123, while it seems fine, please make sure you have a sign-off or an approval from another committer, which is a standard approach. At least, you can wait for some positive comments.

Or, please clarify the reason why you're merging it without any sign-off from other people. In this way, you can a) avoid unexpected mistakes, and b) have another person who can take over your change, say, if you happen to leave this project; otherwise, no one can take over if most of changes are just merged without peer-review.

Although it is legitimate to merge as a committer as far as I know, practically we should better stick to this practice.

**Comment by HyukjinKwon** [comment]
Also, please clarify which branch you merged to avoid the overhead for people to check commit history to see which branch it was merged.

**Review by srowen — COMMENTED** [review]
Looks OK to me.
Yeah I'd say it's a judgment call, if it's a minor change and you feel sure about it and nobody has weighed in for a day or two, it's not unreasonable to proceed. If in doubt I'd try pinging someone for a second look.

**Comment by srowen** [comment]
For the record looks like this was merged to master

**Pull request closed without merging**